sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientTests.cs

```diff
@@ -43,6 +43,46 @@ public void CanSetServiceTimeout_Mocked()
             Assert.AreEqual(TimeSpan.FromMinutes(10).Add(TimeSpan.FromSeconds(15)), networkOverride);
         }
 
+        [Test]
+        public void CanSetServiceTimeoutForBatch_Mocked()
+        {
+            string preferHeader = null;
+            TimeSpan? networkOverride = default;
+
+            var mockTransport = MockTransport.FromMessageCallback(message =>
+            {
+                Assert.True(message.Request.Headers.TryGetValue("prefer", out preferHeader));
+                networkOverride = message.NetworkTimeout;
+
+                return new MockResponse(403);
+            });
+
+            var client = new LogsQueryClient(new Uri("https://api.loganalytics.io"), new MockCredential(), new LogsQueryClientOptions()
+            {
+                Transport = mockTransport
+            });
+
+            var batch = new LogsBatchQuery();
+            batch.AddQuery("wid", "tid", TimeSpan.FromDays(1), options: new LogsQueryOptions()
+            {
+                ServerTimeout = TimeSpan.FromMinutes(1)
+            });
+            batch.AddQuery("wid", "tid", TimeSpan.FromDays(1), options: new LogsQueryOptions()
+            {
+                ServerTimeout = TimeSpan.FromMinutes(2)
+            });
+            batch.AddQuery("wid", "tid", TimeSpan.FromDays(1), options: new LogsQueryOptions()
+            {
+                ServerTimeout = TimeSpan.FromMinutes(3)
+            });
+
+            Assert.ThrowsAsync<RequestFailedException>(() => client.QueryBatchAsync(batch));
+
+            Assert.AreEqual("wait=180", preferHeader);
```

Comment:
why `180`?

Reply:
3 minutes :) Or are you saying I should add a comment?

Reply:
Even better than a comment would be to reference the value that enforces the 3 minutes. Is it defined somewhere or is it configured as an option? If it's a constant value defined by the service, could it ever change?

Reply:
We take MAX of individual query timeouts so it's based on test data.

Reply:
sweet - should we calculate the value in the Assert then?